lib/brainstem/query_strategies/base_strategy.rb

```diff
@@ -29,7 +29,7 @@ def evaluate_scope(scope)
       def evaluate_count(count_scope)
         return primary_presenter.evaluate_count(count_scope) if delegate_count_to_presenter?
 
-        ret = @last_count || count_scope.count
+        ret = @last_count || @options[:paginator]&.get_count(count_scope) || count_scope.reorder(nil).count
```

Comment:
No one: Say you don't like null-coalescing without saying you don't like null-coalescing.
Nate:
```suggestion
        ret = begin
          return @last_count unless @last_count.nil?
          return paginator_override(count_scope) unless paginator_override.nil?
          count_scope.reorder(nil).count
        end
        @last_count = nil
        ret
      end
      
      def paginator_override
        @options[:paginator]
      end
```

Reply:
Also why did we have to add the `reorder(nil)` to the `count_scope`?

Reply:
I feel like the coalescing pattern is very idiomatic ruby - is there some downside in using it that I am unaware of?

Also this suggestion is a change of behavior such that `@last_count` will no longer always be set to nil. The `return` statements will return from the method, not the begin block.

Reply:
I don't have a ton of context here so take this with a grain of salt, but if we have the idea of a `Paginator` that someone can provide, should we just have a default `Paginator` that Brainstem defines that has the same interface? (It's implementation would just be the `@last_count || count_scope.count` stuff). I think that may reveal intent and clean up the code a bit.

Reply:
@brandonduff - That's the thing I really wanted to reach for when starting this effort. I hit a lot of roadblocks along the way, and never revisited it once I understood the problem a bit better. I'll see if I can make something like that work.

Reply:
@ncperry - `reorder(nil)` will make sure that any ordering applied to the scope is removed, since ordering is not needed for a count query.